timemachine/cpp/src/local_md_potentials.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+#include "device_buffer.hpp"
+#include "pinned_host_buffer.hpp"
+#include <memory>
+#include <vector>
+
+#include "bound_potential.hpp"
+#include "curand.h"
+#include "flat_bottom_bond.hpp"
+#include "local_md_utils.hpp"
+
+namespace timemachine {
+
+class LocalMDPotentials {
+
+public:
+    LocalMDPotentials(const int N, const std::vector<std::shared_ptr<BoundPotential>> bps);
+
+    ~LocalMDPotentials();
+
+    DeviceBuffer<unsigned int> *get_free_idxs();
```

Comment:
ugh - this is a bit gnarly, why do we return a raw pointer here? Do we not support returning by const reference? (eg.

```const DeviceBuffer<unsigned int> &get_free_idxs();```

Reply:
If we pass the const reference, the underlying memory gets cleaned up and triggers an error. Could return raw pointer like our other APIs for consistency